ci/docker/linux-apt-r-3.6.dockerfile

```diff
@@ -101,13 +71,5 @@ ENV ARROW_DEPENDENCY_SOURCE=SYSTEM \
     ARROW_INSTALL_NAME_RPATH=OFF \
     ARROW_WITH_BZ2=OFF \
     ARROW_WITH_ZSTD=OFF \
-    ARROW_R_DEV=TRUE
-
-ENV ARROW_BUILD_BENCHMARKS=off
-ENV ARROW_BUILD_EXAMPLES=off
-ENV ARROW_BUILD_TESTS=off
-ENV ARROW_BUILD_UTILITIES=off
-
-# build, install, test R package
-CMD ["/bin/bash", "-c", "/arrow/ci/docker_build_cpp.sh && \
-    /arrow/ci/docker_build_r.sh"]
+    ARROW_R_DEV=TRUE \
+    MAKEFLAGS=-j8
```

Comment:
Again, this should be tuned to what's available on the system, and you should retain the various "off" flags below.